var/spack/repos/builtin/packages/rose/package.py

```diff
@@ -4,79 +4,288 @@
+    # --------------------------------------------------------------------------
+    # Dependencies
+    # --------------------------------------------------------------------------
+    depends_on("autoconf@2.69:", type="build")
+    depends_on("automake@1.14:", type="build")
+    depends_on("libtool@2.4:", type="build")
+    depends_on("boost@1.56.0:", when="~cxx11")
 
-    depends_on("autoconf@2.69", type='build')
-    depends_on("automake@1.14", type='build')
-    depends_on("libtool@2.4", type='build')
-    depends_on("boost@1.47.0:")
+    # C++11 compatible boost and gcc versions required for +cxx11 variant:
+    # https://github.com/spack/spack/wiki/Telcon%3A-2015-05-14
+    depends_on("boost@1.60.0: cxxstd=11", when="+cxx11")
 
-    variant('tests', default=False, description='Build the tests directory')
+    # TODO: Doesn't seem to be a way to require a specific compiler: https://github.com/spack/spack/issues/896
+    # https://www.gnu.org/software/gcc/projects/cxx-status.html#cxx11
+    # depends_on("%gcc@4.8.1:", when="^boost@1.60.0: cxxstd=11")
 
-    variant('binanalysis', default=False, description='Enable binary analysis tooling')
-    depends_on('libgcrypt', when='+binanalysis', type='build')
-    depends_on('py-binwalk', when='+binanalysis', type='run')
+    # --------------------------------------------------------------------------
+    # Variants
+    # --------------------------------------------------------------------------
+    variant("debug", default=False, description="Enable compiler debugging symbols")
+    variant("optimized", default=False, description="Enable compiler optimizations")
 
-    variant('c', default=True, description='Enable c language support')
-    variant('cxx', default=True, description='Enable c++ language support')
+    variant("tests", default=False, description="Build the tests directory")
+    variant("tutorial", default=False, description="Build the tutorial directory")
 
-    variant('fortran', default=False, description='Enable fortran language support')
+    variant(
+        "mvapich2_backend",
+        default=False,
+        description="Enable mvapich2 backend compiler",
+    )
+    depends_on("mvapich2", when="+mvapich2_backend")
 
-    variant('java', default=False, description='Enable java language support')
-    depends_on('java', when='+java')
+    variant("binanalysis", default=False, description="Enable binary analysis tooling")
+    depends_on("libgcrypt", when="+binanalysis", type="build")
+    depends_on("py-binwalk", when="+binanalysis", type="run")
 
-    variant('z3', default=False, description='Enable z3 theorem prover')
-    depends_on('z3', when='+z3')
+    variant("c", default=True, description="Enable c language support")
+    variant("cxx", default=True, description="Enable c++ language support")
 
-    build_directory = 'spack-build'
+    # Use spack install cxxflags=-std=c++11
+    variant("cxx11", default=False, description="Enable c++11 language support")
+
```

Comment:
Remove these extra blank lines?

Reply:
This change is from an older version of the rose package to the one that is currently in the repository.